### DISCUSSION TITLE
Is there a built-in way to close the completion menu when I'm typing and open it when I stop typing?

### DISCUSSION THREAD

**Discussion by heygarrett** [hrsh7th/nvim-cmp#1715] [Q&A]
I'm trying to split the difference between autocompletion and manual completion. I don't want to see the completion menu if I know what I want to type. But if I stop to think, I'd then want the completion menu to appear.

If it makes any difference I'm using `view = { entries = "native" }` in my config, but that's not a hard requirement.

**Comment by gegoune** [comment]
You could disable auto completion and call it from cursor hold auto command maybe?

**Reply by heygarrett** [reply]
Thanks for the suggestion! I tried playing around with this as well as `vim.defer_fn` alongside a `TextChangedP` autocmd to close the completion menu:

```lua
local group = vim.api.nvim_create_augroup("nvim-cmp", { clear = true })
vim.api.nvim_create_autocmd("TextChangedP", {
	group = group,
	callback = function() cmp.close() end,
})
vim.api.nvim_create_autocmd("CursorHoldI", {
	group = group,
	callback = function() cmp.complete() end,
})
```

But for some reason the completion menu closes on its own as soon as it opens. 🤔
